frontend/src/components/TableOfContentsDialog.tsx

```diff
@@ -0,0 +1,94 @@
+import { FC, useState } from 'react'
+
+import TocIcon from '@mui/icons-material/Toc'
+import {
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  Divider,
+  Fab,
+  List,
+  ListItem,
+  ListItemButton,
+  ListItemText,
+  useMediaQuery,
+  useTheme,
+} from '@mui/material'
+import { useTranslation } from 'next-i18next'
+
+import { TableOfContentItem } from '../lib/types'
+
+export interface TableOfContentsDialogProps {
+  header?: string
+  items: ReadonlyArray<TableOfContentItem>
+}
+
+export const TableOfContentsDialog: FC<TableOfContentsDialogProps> = ({
+  header,
+  items,
+}) => {
+  const { t } = useTranslation('common')
+  const [open, setOpen] = useState(false)
+  const theme = useTheme()
+  const fullScreen = useMediaQuery(theme.breakpoints.down('md'))
+
+  const handleOnTriggerClick = () => {
+    setOpen(true)
+  }
+
+  const handleOnDialogClose = () => {
+    setOpen(false)
+  }
+
+  const handleOnCloseClick = () => {
+    setOpen(false)
+  }
+
+  const handleListItemClick = () => {
+    setOpen(false)
+  }
```

Comment:
Why extract these out as variables when they can be inlined directly in the react components below?

Reply:
Because of this and I'm used to do it. Might use function instead of array function too. 🤷 
https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-no-bind.md

Reply:
I guess that makes sense.. I don't think it applies in this exact situation, but for code consistency it makes sense to always stick to one approach.

I'll begrudgingly resolve. :hankey: